src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -372,12 +372,13 @@ private function parseGroups(string $regex): ?array
 	{
 		if (self::$parser === null) {
 			/** @throws void */
-			self::$parser = Llk::load(new Read('hoa://Library/Regex/Grammar.pp'));
+			self::$parser = Llk::load(new Read(__DIR__ . '/../../../resources/RegexGrammar.pp'));
```

Comment:
@ondrejmirtes Are we ok loading the file from here instead? Not sure what this does in terms of licensing, nor if there's a better place to put it. If so I'll delete the patch file and drop the hoa/regex requirement.

Reply:
I am not sure this path will work in the packaged phpstan.phar

Reply:
I completely agree moving from the patching as the patching is very hard to understand and the changes needed are more and more complex.

Reply:
Ok I got rid of it, we'll see what @ondrejmirtes says :)

Reply:
I propose to add the whole and unpatched `RegexGrammar.pp` as first/separate commit, then the patched/actual grammar and then the changes from this PR. This will hugely improve the review and git history.

Reply:
Good idea, done 👍🏻